test/Microsoft.AspNetCore.Mvc.Razor.Extensions.Test/PageDirectiveTest.cs

```diff
@@ -0,0 +1,127 @@
+    {
+        [Fact]
+        public void TryGetPageDirective_ReturnsFalse_IfPageDoesNotHaveDirective()
+        {
+            // Arrange
+            var content = "Hello world";
+            var sourceDocument = RazorSourceDocument.Create(content, "file");
+            var codeDocument = RazorCodeDocument.Create(sourceDocument);
+            var engine = CreateEngine();
+            var irDocument = CreateIRDocument(engine, codeDocument);
+
+            // Act
+            var result = PageDirective.TryGetPageDirective(irDocument, out var pageDirective);
+
+            // Assert
+            Assert.False(result);
+        }
+
+        [Fact(Skip = "https://github.com/aspnet/Razor/issues/1201")]
+        public void TryGetPageDirective_ReturnsFalse_IfPageDoesStartWithDirective()
+        {
+            // Arrange
+            var content = "Hello @page";
+            var sourceDocument = RazorSourceDocument.Create(content, "file");
+            var codeDocument = RazorCodeDocument.Create(sourceDocument);
+            var engine = CreateEngine();
+            var irDocument = CreateIRDocument(engine, codeDocument);
+
+            // Act
+            var result = PageDirective.TryGetPageDirective(irDocument, out var pageDirective);
+
+            // Assert
+            Assert.False(result);
+        }
+
+        [Fact]
+        public void TryGetPageDirective_ReturnsTrue_IfContentHasDirective()
+        {
+            // Arrange
+            var content = "@page";
+            var sourceDocument = RazorSourceDocument.Create(content, "file");
+            var codeDocument = RazorCodeDocument.Create(sourceDocument);
+            var engine = CreateEngine();
+            var irDocument = CreateIRDocument(engine, codeDocument);
+
+            // Act
+            var result = PageDirective.TryGetPageDirective(irDocument, out var pageDirective);
+
+            // Assert
+            Assert.True(result);
+            Assert.Null(pageDirective.RouteTemplate);
+            Assert.Null(pageDirective.PageName);
+        }
+
+        [Fact]
+        public void TryGetPageDirective_ParsesRouteTemplate()
+        {
+            // Arrange
+            var content = "@page \"some-route-template\"";
+            var sourceDocument = RazorSourceDocument.Create(content, "file");
+            var codeDocument = RazorCodeDocument.Create(sourceDocument);
+            var engine = CreateEngine();
+            var irDocument = CreateIRDocument(engine, codeDocument);
+
+            // Act
+            var result = PageDirective.TryGetPageDirective(irDocument, out var pageDirective);
+
+            // Assert
+            Assert.True(result);
+            Assert.Equal("some-route-template", pageDirective.RouteTemplate);
+            Assert.Null(pageDirective.PageName);
+        }
+
+        [Fact]
+        public void TryGetPageDirective_ParsesPageName()
+        {
+            // Arrange
+            var content = "@page \"some-route\" \"some name\"";
```

Comment:
Can we get a test that `@page "some-route" non-razor garbage` doesn't give you "non-razor garbage" as pageDirective.PageName?